### ISSUE TITLE
Cop idea: Snake Case for Symbols

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12346]
## Description

Use `snake_case` for symbols.

https://rubystyle.guide/#snake-case-symbols-methods-vars

## Describe the solution you'd like

```ruby
# bad
:'some symbol'
:SomeSymbol
:someSymbol
%i[SomeSymbol]

# good
:some_symbol
%i[some_symbol]
```

**Comment by fatkodima** [comment]
Already implemented. You can find these cops by searching for `snake-case-symbols-methods-vars` in the `default.yml` file.

**Comment by ydakuka** [comment]
I've just checked it again.

I have the code:

```ruby
class Test
  def my_method
    [:'some symbol', :SomeSymbol, :someSymbol, %i[SomeSymbol]]
  end
end
```

Terminal:
```bash
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop app/models/test.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```
```bash
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop app/models/test.rb --only Naming/MethodName,Naming/VariableName,Naming/VariableNumber
Inspecting 1 file
.

1 file inspected, no offenses detected
```

**Comment by fatkodima** [comment]
Yeah, sorry, symbols are missing. It is currently implemented only for method names and variable names.

**Comment by koic** [comment]
IMHO, the style guide might have overlooked this. While method names and variable names can be dictated by coding rules, symbols tend to lean more towards strings, which aren't typically enforced by such regulations.

For instance, one might consider CamelCase representations for external parameters. However, it's not feasible to universally define such symbols within a coding rule.

Alternatively, the style guide serves as a guideline, and there are cases where it shouldn't necessarily be enforced uniformly through a cop. So, the guide isn't wrong, but this might not be a case to implement as a cop. This seems to be one of those cases.
